src/CLITests/ErrorCollectorUnitTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using AxeWindowsScanner;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace CLITests
+{
+    [TestClass]
+    public class ErrorCollectorUnitTests
+    {
+        const string TrivialValue = "   \n   \n\t\n";
+        const string ValidValue = "Some\n Valid\n Value";
+        static readonly string[] ValidValueArray = { "Value 1", "Value 2", "Value 3" };
+
+        IErrorCollector _errorCollector;
+
+        [TestInitialize]
+        public void BeforeEachTest()
+        {
+            _errorCollector = new ErrorCollector();
+        }
+
+        private void ValidateListSizes(int expectedParameterErrors = 0, int expectedExceptions = 0)
+        {
+            Assert.AreEqual(expectedParameterErrors, _errorCollector.ParameterErrors.Count);
+            Assert.AreEqual(expectedExceptions, _errorCollector.Exceptions.Count);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void ValidInitialListSizes()
+        {
+            ValidateListSizes();
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void AddParameterError_ValueIsTrivial_ThrowsArgumentException()
+        {
+            ArgumentException e = Assert.ThrowsException<ArgumentException>(
+                () => _errorCollector.AddParameterError(TrivialValue));
+            Assert.AreEqual("error", e.ParamName);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void AddParameterError_ValueIsValid_StoresValueCorrectly()
+        {
+            _errorCollector.AddParameterError(ValidValue);
+            Assert.AreEqual(ValidValue, _errorCollector.ParameterErrors[0]);
+            ValidateListSizes(expectedParameterErrors: 1);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void AddParameterError_MutipleValidValues_StoresValuesCorrectly()
+        {
+            foreach (string value in ValidValueArray)
+            {
+                _errorCollector.AddParameterError(value);
+            }
+
+            for (int loop = 0; loop < ValidValueArray.Length; loop++)
```

Comment:
I'm curious about the use of "loop" instead of 'i'.

Reply:
Personal preference, I guess. I find "i" much less descriptive than "loop" or "index" (depending on the specific circumstance)